x/evm/keeper/baseKeeper.go

```diff
@@ -49,24 +45,67 @@ func (k BaseKeeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
 }
 
+// InitChains initializes all existing EVM chains and their respective param subspaces
+func (k *BaseKeeper) InitChains(ctx sdk.Context) {
+	if k.initialized {
+		panic("chains are already initialized")
+	}
+
+	iter := k.getBaseStore(ctx).IteratorNew(key.FromStr(subspacePrefix))
+	defer utils.CloseLogError(iter, k.Logger(ctx))
+
+	for ; iter.Valid(); iter.Next() {
+		_ = k.getSubspace(ctx, nexus.ChainName(iter.Key()))
+	}
+
+	k.initialized = true
+}
+
+// CreateChain creates the subspace for a new EVM chain. Returns an error if the chain already exists
+func (k BaseKeeper) CreateChain(ctx sdk.Context, params types.Params) error {
+	if !k.initialized {
+		panic("InitChain must be called before chain keepers can be used")
+	}
+
+	chainKey := key.FromStr(subspacePrefix).Append(key.FromStr(params.Chain.String()))
+	if k.getBaseStore(ctx).HasNew(chainKey) {
+		return fmt.Errorf("chain %s already exists", params.Chain)
+	}
+
+	k.getBaseStore(ctx).SetRawNew(chainKey, []byte(params.Chain))
+
+	subspace := k.getSubspace(ctx, params.Chain)
+	subspace.SetParamSet(ctx, &params)
+	return nil
+}
+
 // ForChain returns the keeper associated to the given chain
-func (k BaseKeeper) ForChain(chain nexus.ChainName) types.ChainKeeper {
-	return chainKeeper{
-		BaseKeeper:    k,
-		chainLowerKey: strings.ToLower(chain.String()),
+func (k BaseKeeper) ForChain(ctx sdk.Context, chain nexus.ChainName) (types.ChainKeeper, error) {
+	if !k.initialized {
+		panic("InitChain must be called before chain keepers can be used")
+	}
+
+	chainKey := key.FromStr(subspacePrefix).Append(key.FromStr(chain.String()))
+	if !k.getBaseStore(ctx).HasNew(chainKey) {
+		return chainKeeper{}, fmt.Errorf("unknown chain %s", chain)
```

Comment:
Does this still enables `ForChain` to be case insensitive?

Reply:
yes, the key is case insensitive